### ISSUE TITLE
A property descriptor must have keys "enumerable" and "configurable"

### ISSUE THREAD

**Issue by zhirzh** [microsoft/TypeScript#35102]
**TypeScript Version:**  3.7.2

**Expected behavior:** In a property descriptor object, keys `configurable` and `enumerable` should not be optional.

**Actual behavior:** These properties are marked as optional (see [here][1] and [here][2]).

[1]: https://github.com/microsoft/TypeScript/blob/v3.7.2/src/lib/es5.d.ts#L88-L89
[2]: https://github.com/microsoft/TypeScript/blob/v3.7.2/src/lib/es5.d.ts#L1368-L1369

**Comment by IllusionMH** [comment]
Why they should be required?
MDM mentions them as optional https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#Description

According to latest draft of spec (and previous versions) https://tc39.es/ecma262/#sec-topropertydescriptor explicitly check if these keys are in descriptor object without throwing if they are not available.

**Comment by zhirzh** [comment]
Yes, you are right on that. I was only comparing the result of `Object.getOwnPropertyDescriptor()` and concluded that keys `configurable` and `enumerable` must be present.

**Comment by RyanCavanaugh** [comment]
The return type of `Object.getOwnPropertyDescriptor` could be more precise by making these non-optional since they're always returned

**Comment by RyanCavanaugh** [comment]
Doesn't seem like this is worth making a type difference for.
